share/functions/__fish_portage_print_repository_names.fish

```diff
@@ -0,0 +1,4 @@
+function __fish_portage_print_repository_names --description 'Print the names of all configured repositories'
+	# repos.conf may be a file or a directory
+	find /etc/portage/repos.conf -type f -exec sed -ne '/^[[:blank:]]*\[[[:alnum:]_][[:alnum:]_-]*\]/{s!^.*\[\(.*\)\].*$!\1!; /DEFAULT/!p}' '{}' +
```

Comment:
What does the input here look like?

Reply:
And what are you trying to extract?

(This goes for print_repository_paths as well)

Reply:
The format is ini-like, as in
```
[repository]
key = value
# comment
```
In this case the part between `[ ]` is extracted, and the special "repository" `DEFAULT` removed from the output.
In the case of print_repository_paths the value of the key "location" is extracted.

Reply:
Leave it like this for now. If the completions turn out to be slow, this is my guess. But I'd assume few people have hundreds of repository files configured.

Reply:
It should be <5 for most people. Additionally, the files themselves are rather small.

Reply:
Also should completion be slow I'd assume it's because gentoo's main repository alone contains ~20k packages.